nemo/collections/tts/models/aligner.py

```diff
@@ -181,31 +182,60 @@ def validation_step(self, batch, batch_idx):
-                        ),
-                        caption=f"attn hard",
-                    )
-                )
-
-            self.logger.experiment.log({"attn_matrices": attn_matrices})
+        for logger in self.loggers:
+            if batch_idx == 0:
+                if isinstance(self.logger, WandbLogger) and HAVE_WANDB:
+                    if attn_hard is None:
+                        attn_hard = binarize_attention(attn_soft, text_len, spec_len)
+
+                    attn_matrices = []
+                    for i in range(min(5, audio.shape[0])):
+                        attn_matrices.append(
+                            wandb.Image(
+                                plot_alignment_to_numpy(
+                                    np.fliplr(
+                                        np.rot90(attn_soft[i, 0, : spec_len[i], : text_len[i]].data.cpu().numpy())
+                                    )
+                                ),
+                                caption=f"attn soft",
+                            ),
+                        )
+
+                        attn_matrices.append(
+                            wandb.Image(
+                                plot_alignment_to_numpy(
+                                    np.fliplr(
+                                        np.rot90(attn_hard[i, 0, : spec_len[i], : text_len[i]].data.cpu().numpy())
+                                    )
+                                ),
+                                caption=f"attn hard",
+                            )
+                        )
+
+                    self.logger.experiment.log({"attn_matrices": attn_matrices})
+
+                elif isinstance(logger, ClearMLLogger) and HAVE_CLEARML_LOGGER:
+                    if attn_hard is None:
+                        attn_hard = binarize_attention(attn_soft, text_len, spec_len)
+
+                    # Attentions:
+                    for i in range(min(5, audio.shape[0])):
+                        logger.clearml_task.logger.report_image(
+                            image=plot_alignment_to_numpy(
+                                np.fliplr(np.rot90(attn_soft[i, 0, : spec_len[i], : text_len[i]].data.cpu().numpy()))
+                            ),
+                            series=f"attn soft {i}",
+                            title="attn_matrices",
+                            iteration=self.global_step,
+                        )
+                        logger.clearml_task.logger.report_image(
+                            image=plot_alignment_to_numpy(
+                                np.fliplr(np.rot90(attn_hard[i, 0, : spec_len[i], : text_len[i]].data.cpu().numpy()))
+                            ),
+                            series=f"attn hard {i}",
+                            title="attn_matrices",
+                            iteration=self.global_step,
+                        )
```

Comment:
But I think it might break code execution if some custom user's logger is passed to the trainer.

1. Maybe we can use this (pseudocode):
```python
for logger in self.loggers:
    ...
    elif not self.supported_loggers_warning_printed:
        logging.warning(f"Logger {logger} is not supported for logging validation samples!")?
self.supported_loggers_warning_printed = True
```
2. Or create class like `TTSValLogger`, from each TTS model class should inherit.
    -  `TTSValLogger` can implement method to check loggers on startup and print warnings about unsupported loggers. 
    -  Also it can implement common function for all TTS models for validation samples logging.

Reply:
Maybe locate this class in new file nemo.collections.tts.parts.utils.loggers.py